narwhal/primary/src/synchronizer.rs

```diff
@@ -125,6 +125,15 @@ impl Inner {
     ) -> DagResult<()> {
         let digest = certificate.digest();
 
+        // TODO: remove this validation later to reduce rocksdb access.
+        if certificate.round() > 1 {
+            for digest in &certificate.header.parents {
+                if !self.certificate_store.contains(digest)? {
```

Comment:
What are your thoughts on doing a `multi_get()` instead to optimize rocksdb access in the interim? (same can be applied in consensus::check_parents)

Reply:
Good point! But this would require a bit more plumbing to certificate store. I may move this parent validation commit to another PR.